### ISSUE TITLE
Can you provide train_val.prototxt for an example amazon to dslr which has 31 classes?

### ISSUE THREAD

**Issue by mahfuj9346449** [yhldhit/WMMD-Caffe#4]
Can you provide train_val.prototxt for an example amazon to dslr which has 31 classes? Because in your example there are some parameters i.,e., batch_size: 1123 ,  num_of_kernel: 10.... what should be for 31 classes?

**Comment by yhldhit** [comment]
We empirically basic learning rate of SGD, the coefficient of weighted MMD regularizor, and the coefficient of empirical loss on target domain. Batchsize is adjusted based on the size of dataset.

**Comment by mahfuj9346449** [comment]
Thanks.

**Comment by mengchuangji** [comment]
请问你31类的可以跑吗？  我的10类的可以跑，但31类的总有问题，虽然能改的地方都改了（比如mmd_layer.cu,train_val.ptototxt）。我的环境是cuda8.0,cudnn3.0,NVIDIA 390.


编译时，会有警告：
src/caffe/layers/softmax_loss_layer.cu: In instantiation of ‘void caffe::SoftmaxWithLossLayer<Dtype>::Forward_gpu(const std::vector<caffe::Blob<Dtype>*>&, const std::vector<caffe::Blob<Dtype>*>&) [with Dtype = float]’:
src/caffe/layers/softmax_loss_layer.cu:231:143:   required from here
src/caffe/layers/softmax_loss_layer.cu:75:36: warning: non-constant array new length must be specified without parentheses around the type-id [-Wvla]
       Dtype * prob_samp = new Dtype[dim];
                                    ^
src/caffe/layers/softmax_loss_layer.cu: In instantiation of ‘void caffe::SoftmaxWithLossLayer<Dtype>::Forward_gpu(const std::vector<caffe::Blob<Dtype>*>&, const std::vector<caffe::Blob<Dtype>*>&) [with Dtype = double]’:
src/caffe/layers/softmax_loss_layer.cu:231:290:   required from here
src/caffe/layers/softmax_loss_layer.cu:75:36: warning: non-constant array new length must be specified without parentheses around the type-id [-Wvla]

运行时会出现：
I0403 17:04:24.901782 10596 net.cpp:638] Copying source layer mmd_fc8
I0403 17:04:24.901789 10596 net.cpp:638] Copying source layer loss
I0403 17:04:25.011085 12030 image_data_layer.cpp:177] Prefetch batch: 124 ms.
I0403 17:04:25.011171 12030 image_data_layer.cpp:178]      Read time: 59.597 ms.
I0403 17:04:25.011207 12030 image_data_layer.cpp:179] Transform time: 64.661 ms.
*** stack smashing detected ***: ./build/tools/caffe terminated
*** Aborted at 1554282265 (unix time) try "date -d @1554282265" if you are using GNU date ***
PC: @     0x7f043a83b428 gsignal
*** SIGABRT (@0x2964) received by PID 10596 (TID 0x7f043c7d3740) from PID 10596; stack trace: ***
    @     0x7f043a83b4b0 (unknown)
    @     0x7f043a83b428 gsignal
    @     0x7f043a83d02a abort
    @     0x7f043a87d7ea (unknown)
    @     0x7f043a91f15c __fortify_fail
    @     0x7f043a91f100 __stack_chk_fail
    @     0x7f043b961f5c caffe::SoftmaxWithLossLayer<>::Forward_gpu()
    @           0x418fd8 caffe::Layer<>::Forward()
    @     0x7f043b8eb243 caffe::Net<>::ForwardFromTo()
    @     0x7f043b8eaf85 caffe::Net<>::ForwardPrefilled()
    @     0x7f043b8eb3ba caffe::Net<>::Forward()
    @     0x7f043b93dccf caffe::Solver<>::Test()
    @     0x7f043b93dadf caffe::Solver<>::TestAll()
    @     0x7f043b93cb41 caffe::Solver<>::Step()
    @     0x7f043b93c72f caffe::Solver<>::Solve()
    @           0x414594 train()
    @           0x4166fb main
    @     0x7f043a826830 __libc_start_main
    @           0x4138f9 _start
    @                0x0 (unknown)
Aborted (core dumped)
然后停止。
